### PR TITLE
Encode strings that look like hexcodes with underscores

### PR DESCRIPTION
**Pull request by lavham**
Fork xlsxtream code to properly respect hex-similar strings by replacing _ via _x005f_

Implements Underscore escape per Microsoft Excel spec:
> Underscore (0x005f): This character shall be escaped only when used to escape the first underscore character in the format _xHHHH_.

### CODE REVIEW

File: lib/xlsxtream/xml.rb [hunk 1]
```diff
@@ -36,8 +39,18 @@ def escape_attr(string)
         string.gsub(UNSAFE_ATTR_CHARS, XML_ESCAPES)
       end
 
+      # Add underscore to strings that merely look like hex values, preventing manipulation into invalid UTF8
```
Comment (hislopzach) [inline lib/xlsxtream/xml.rb]: 🔧  Replace this first comment with the following:
> Ensure that we escape the first underscore character in plaintext strings that match the format for Excel escape sequences. This ensures that these strings are displayed as plaintext and not incorrectly parsed as escape sequences by Excel

File: lib/xlsxtream/xml.rb [hunk 2]
```diff
@@ -36,8 +39,18 @@ def escape_attr(string)
...
+      # Per Microsoft Open Specifications for Excel:
+      # Underscore (0x005f): This character shall be escaped only when used to escape the first underscore character in the format _xHHHH_.
+      def encode_underscores_using_x005f(string)
```
Comment (hislopzach) [inline lib/xlsxtream/xml.rb]: 🔧 function name suggestion: `ensure_strings_that_match_escape_sequence_are_escaped`
a bit wordy, but I feel that it concisely captures the intent of the method

File: test/xlsxtream/xml_test.rb [hunk 3]
```diff
@@ -42,5 +42,22 @@ def test_escape_value_valid_xml_chars
...
+    def test_encode_underscores_using_x005f_multiple_occurrences
+      unsafe_value = "_xDcc2_aa_x3d12_bb_xDea3_cc_xDaa5_"
+      safe_value = "_x005f_xDcc2_aa_x005f_x3d12_bb_x005f_xDea3_cc_x005f_xDaa5_"
+      assert_equal safe_value, XML.escape_value(unsafe_value)
+    end
+
+    def test_not_escaping_regular_underscores
+      safe_value = "this_test_does_not_replace_underscores_xDcc2"
+      assert_equal safe_value, XML.escape_value(safe_value)
+    end
```
Comment (hislopzach) [inline test/xlsxtream/xml_test.rb:61]: ❓ do we want unit tests for the method we added? we are getting the coverage by testing the method that calls our method. Unsure if there are any tests that would be different though

**Review by hislopzach — CHANGES_REQUESTED** [review]

File: lib/xlsxtream/xml.rb (hunk 2 above)
Comment (lavham) [reply lib/xlsxtream/xml.rb]: What about `escape_strings_that_match_excel_escape_sequence`?

File: test/xlsxtream/xml_test.rb (hunk 3 above)
Comment (lavham) [reply test/xlsxtream/xml_test.rb:61]: Nah I think it's ok given we care about the outcome of `escape_value` and the tests would be identical to your point.

**Review by hislopzach — APPROVED** [review]
🚀

File: lib/xlsxtream/xml.rb [hunk 4]
```diff
@@ -36,8 +39,19 @@ def escape_attr(string)
         string.gsub(UNSAFE_ATTR_CHARS, XML_ESCAPES)
       end
 
+      # Ensure that we escape the first underscore character in plaintext strings that match the format for Excel escape sequences.
+      # This ensures that these strings are displayed as plaintext and not incorrectly parsed as escape sequences by Excel
+      # Per Microsoft Open Specifications for Excel:
```
Comment (hislopzach) [inline lib/xlsxtream/xml.rb]: I just approved, but I think we should link to the spec doc in this comment. Makes it clear what 'spec' we're referencing, and will help future devs of this gem once this is merged to the main gem

Comment (lavham) [reply lib/xlsxtream/xml.rb]: Updated: https://github.com/Invoca/xlsxtream/pull/1/commits/2e39341c4b8f8a60075ff185aa318895ca806ad4
